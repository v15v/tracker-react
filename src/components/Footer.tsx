import React from "react";

interface Props {
    footer: {
        title: string,
        subtitle: string,
        copyright: string
    }
}

const Footer = ({footer}: Props) => (
    <section className="section has-background-info">
        <div className="container is-widescreen">
            <p className="is-size-7 has-text-centered has-text-light">
                {footer ? footer.copyright : ""}
            </p>
        </div>
    </section>
)

export default Footer
