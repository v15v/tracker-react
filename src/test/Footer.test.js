import {cleanup, render, screen} from '@testing-library/react';
import '@testing-library/jest-dom'
import Footer from "../components/Footer"

// Note: running cleanup afterEach is done automatically for you in @testing-library/react@9.0.0 or higher
// unmount and cleanup DOM after the test is finished.
afterEach(cleanup);


describe('Footer', () => {
    it("renders the Footer copyright", () => {
        const footer = {copyright: "Footer"}
        render(<Footer footer={footer}/>)

        expect(screen.getByText(/Footer/)).toBeInTheDocument()
    })
    it("renders the Footer other copyright", () => {
        const footer = {copyright: "Some copyright"}
        render(<Footer footer={footer}/>)

        expect(screen.getByText(/Some copyright/)).toBeInTheDocument()
    })
    it("renders the Footer with copyright only", () => {
        const footer = {copyright: "Footer copyright"}
        render(<Footer footer={footer}/>)

        expect(screen.getByText(/Footer copyright/)).toBeInTheDocument()
    })
    it('Footer snapshot', () => {
        const footer = {copyright: "Footer"}
        const view = render(<Footer footer={footer}/>)

        expect(view).toMatchSnapshot()
    })
    it('Footer without data snapshot', () => {
        const view = render(<Footer/>)

        expect(view).toMatchSnapshot()
    })
})