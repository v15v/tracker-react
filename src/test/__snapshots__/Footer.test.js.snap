// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Footer snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="section has-background-info"
      >
        <div
          class="container is-widescreen"
        >
          <p
            class="is-size-7 has-text-centered has-text-light"
          >
            Footer
          </p>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="section has-background-info"
    >
      <div
        class="container is-widescreen"
      >
        <p
          class="is-size-7 has-text-centered has-text-light"
        >
          Footer
        </p>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Footer Footer without data snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="section has-background-info"
      >
        <div
          class="container is-widescreen"
        >
          <p
            class="is-size-7 has-text-centered has-text-light"
          />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="section has-background-info"
    >
      <div
        class="container is-widescreen"
      >
        <p
          class="is-size-7 has-text-centered has-text-light"
        />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
